crossopt/assignment1/Q2.py

```diff
@@ -33,14 +33,17 @@ def __str__(self):
             cnt = cnt.next
         return "{" + ", ".join(ans) + "}"
 
-    # return kth element from end of list
-    def getKth(self, k):
-        len = self.length()
-        if k >= len:
+    # list[-k] returns kth element from end of list, last element is -1
+    def __getitem__(self, k):
+        length = len(self)
+
+        if k >= 0:
+            raise NotImplementedError()
+        elif k < -length:
             raise IndexError()
 
         cnt = self.head
-        for i in range(len - k - 1):
+        for i in range(length + k):
             cnt = cnt.next
         return cnt
 
```

Comment:
For the second assignement, I will require that you use unit tests :)

You can take a look at my pull request or the pull request of the other mentees:
Laida:  https://github.com/flerdacodeu/CodeU-2018-Group2/commit/14499bcd73a276f0ae1686a6b4fa6cbb3483a37f
JD: https://github.com/flerdacodeu/CodeU-2018-Group2/pull/10/files